### PR TITLE
Add option to use a remote debugger proxy

### PR DESCRIPTION
**Pull request by ThomasFettSauce**
All further info regarding this pull request can be found at https://github.com/appium/appium/pull/10960

### CODE REVIEW

File: lib/remote-debugger-rpc-client.js [hunk 1]
```diff
@@ -44,10 +48,27 @@ export default class RemoteDebuggerRpcClient {
 
     // create socket and handle its messages
     if (this.socketPath) {
-      // unix domain socket
-      log.debug(`Connecting to remote debugger through unix domain socket: '${this.socketPath}'`);
-      this.socket = net.connect(this.socketPath);
+      if (this.messageProxySocketPath) {
+        // unix domain socket via proxy
+        log.debug(`Connecting to remote debugger through unix domain socket: '${this.messageProxySocketPath}'`);
```
Comment (imurchie) [inline lib/remote-debugger-rpc-client.js]: I think there should be some indication in the logs, here and below, that this is being proxied.

File: lib/remote-debugger-rpc-client.js [hunk 2]
```diff
@@ -44,10 +48,27 @@ export default class RemoteDebuggerRpcClient {
...
+        this.socket = net.connect(this.messageProxySocketPath);
+
+        // Forward the actual socketPath to the proxy
+        this.socket.once('connect', () => {
```
Comment (mykola-mokhnach) [inline lib/remote-debugger-rpc-client.js]: shouldn't the event handler be assigned before calling connect?

Comment (imurchie) [reply lib/remote-debugger-rpc-client.js]: That is not possible. The socket is created by `net.connect`.

File: lib/remote-debugger-rpc-client.js (hunk 1 above)
Comment (ThomasFettSauce) [reply lib/remote-debugger-rpc-client.js]: Good point! I adjusted the logs.

**Comment by imurchie** [comment]
Published in `appium-remote-debugger@3.14.0`.

**Pull request merged**